.github/workflows/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+name: "Python"
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - '**'
+
+jobs:
+  lint:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8==5.0.4 flake8-import-order==0.18.1 darglint==1.8.1 codespell mypy types-requests
+
+      - name: Lint with flake8
+        run: |
+          flake8 --statistics
+
+      - name: Check static typing
+        run: |
+          mypy . --explicit-package-bases || true
```

Comment:
Maybe nice to enable this check? We only need to add type hints to a couple of lines to make it pass.